plugins/generic/recommendBySimilarity/RecommendBySimilarityPlugin.php

```diff
@@ -8,104 +8,130 @@
+            });
         }
-        return $success;
+        return true;
     }
 
     /**
-     * @see Plugin::getDisplayName()
+     * Builds the template with the recommended submissions or null if the linked submission has no keywords
+     *
+     * @see templates/article/footer.tpl
      */
-    public function getDisplayName()
+    private function buildTemplate(): ?string
     {
-        return __('plugins.generic.recommendBySimilarity.displayName');
+        $templateManager = TemplateManager::getManager();
+        $submissionId = $templateManager->getTemplateVars('article')->getId();
+
+        // If there's no keywords, quit
+        if (!strlen($searchPhrase = implode(' ', (new ArticleSearch())->getSimilarityTerms($submissionId)))) {
+            return null;
+        }
+
+        $request = Application::get()->getRequest();
+        $router = $request->getRouter();
+        $context = $router->getContext($request);
+
+        $rangeInfo = Handler::getRangeInfo($request, 'articlesBySimilarity');
+        $rangeInfo->setCount(static::DEFAULT_RECOMMENDATION_COUNT);
+
+        // Prepares the collector to retrieve similar submissions
+        $collector = Repo::submission()
+            ->getCollector()
+            ->excludeIds([$submissionId])
+            ->filterByContextIds([$context->getId()])
+            ->filterByStatus([Submission::STATUS_PUBLISHED])
+            ->searchPhrase($searchPhrase);
+
+        $offset = ($rangeInfo->getPage() - 1) * $rangeInfo->getCount();
+        $submissionCount = $collector->getCount();
+
+        $submissions = $collector
+            ->limit($rangeInfo->getCount())
+            ->offset($offset)
+            ->orderBy(Collector::ORDERBY_SEARCH_RANKING)
+            ->getMany();
+
+        // Load the linked issues
+        $issues = Repo::issue()->getCollector()
+            ->filterByContextIds([$context->getId()])
+            ->filterByIssueIds(
+                $submissions->map(fn (Submission $submission) => $submission->getCurrentPublication()->getData('issueId'))
+                    ->unique()
+                    ->toArray()
+            )
+            ->getMany();
+
+        $nextPage = $rangeInfo->getPage() * $rangeInfo->getCount() < $submissionCount
+            ? $request->url(path: $submissionId, params: ['articlesBySimilarityPage' => $rangeInfo->getPage() + 1])
+            : null;
+        $previousPage = $rangeInfo->getPage() > 1
+            ? $request->url(path: $submissionId, params: ['articlesBySimilarityPage' => $rangeInfo->getPage() - 1])
+            : null;
+
+        $templateManager->assign('articlesBySimilarity', (object) [
+            'submissions' => $submissions,
+            'query' => $searchPhrase,
+            'issues' => $issues,
+            'start' => $offset + 1,
+            'end' => $offset + $submissions->count(),
+            'total' => $submissionCount,
+            'nextUrl' => $nextPage,
+            'previousUrl' => $previousPage
```

Comment:
I'm passing all these variables to fill the `frontend/components/pagination.tpl`, perhaps this can be transformed into a Smarty tag 🤔